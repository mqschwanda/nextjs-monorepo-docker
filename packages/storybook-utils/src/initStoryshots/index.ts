import type { StoryshotsOptions } from '@storybook/addon-storyshots/dist/api/StoryshotsOptions';
import initAddonStoryshots from '@storybook/addon-storyshots';

/**
 * Options for the `initStoryshots` function.
 */
export type InitStoryshotsOptions = StoryshotsOptions & {
  /**
   * If the snapshot tests fail when react logs a key error.
   *
   * Set this to `false` if stories avoid using the key because it would show up in the code example.
   *
   * @default true
   */
  failOnReactKeyError?: boolean
};

/**
 * A wrapper for the `@storybook/addon-storyshots` module
 *
 * See [addon storyshots docs](https://storybook.js.org/addons/@storybook/addon-storyshots) for more information.
 */
export default function initStoryshots({
  failOnReactKeyError = true,
  ...options
}: InitStoryshotsOptions) {
  if (failOnReactKeyError === false) {
    const consoleError = jest.spyOn(console, 'error');

    beforeAll(() => {
      consoleError.mockImplementation((...args) => {
        const [message] = args;
        if (message.includes('Each child in a list should have a unique "key" prop.')) {
          return;
        }

        console.error(...args); // eslint-disable-line no-console
      });
    });

    afterAll(() => {
      consoleError.mockRestore();
    });
  }

  initAddonStoryshots(options);
}
