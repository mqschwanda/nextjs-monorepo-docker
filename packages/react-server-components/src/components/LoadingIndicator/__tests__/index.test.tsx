import { render } from '@testing-library/react';
import initStoryshots from '@/__tests__/initStoryshots';
import { LoadingIndicator } from '@';

initStoryshots({
  failOnReactKeyError: false,
  storyKindRegex: /^@mqs\/react-server-components\/components\/LoadingIndicator$/,
});

describe('@mqs/react-server-components', () => {
  describe('components', () => {
    describe('<LoadingIndicator />', () => {
      it('renders', () => {
        const testId = LoadingIndicator.name;

        const { getByTestId, asFragment } = render((
          <LoadingIndicator
            className='test'
            testId={testId}
          />
        ));

        const element = getByTestId(testId);
        expect(element).toBeTruthy();

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
