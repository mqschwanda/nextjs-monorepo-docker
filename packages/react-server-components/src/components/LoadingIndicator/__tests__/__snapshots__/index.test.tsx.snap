// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mqs/react-server-components components <LoadingIndicator /> renders 1`] = `
<DocumentFragment>
  <span
    class="loading loading-spinner loading-md test"
    data-testid="LoadingIndicator"
  />
</DocumentFragment>
`;

exports[`Storyshots @mqs/react-server-components/components/LoadingIndicator Loading Indicator Color Example 1`] = `
[
  <span
    className="loading text-primary loading-spinner loading-md"
  >
    primary
  </span>,
  <span
    className="loading text-secondary loading-spinner loading-md"
  >
    secondary
  </span>,
  <span
    className="loading text-accent loading-spinner loading-md"
  >
    accent
  </span>,
  <span
    className="loading text-neutral loading-spinner loading-md"
  >
    neutral
  </span>,
  <span
    className="loading text-info loading-spinner loading-md"
  >
    info
  </span>,
  <span
    className="loading text-success loading-spinner loading-md"
  >
    success
  </span>,
  <span
    className="loading text-warning loading-spinner loading-md"
  >
    warning
  </span>,
  <span
    className="loading text-error loading-spinner loading-md"
  >
    error
  </span>,
]
`;

exports[`Storyshots @mqs/react-server-components/components/LoadingIndicator Loading Indicator Example 1`] = `
<span
  className="loading loading-spinner loading-md"
/>
`;

exports[`Storyshots @mqs/react-server-components/components/LoadingIndicator Loading Indicator Shape Example 1`] = `
[
  <span
    className="loading loading-ball loading-md"
  >
    ball
  </span>,
  <span
    className="loading loading-bars loading-md"
  >
    bars
  </span>,
  <span
    className="loading loading-dots loading-md"
  >
    dots
  </span>,
  <span
    className="loading loading-infinity loading-md"
  >
    infinity
  </span>,
  <span
    className="loading loading-ring loading-md"
  >
    ring
  </span>,
  <span
    className="loading loading-spinner loading-md"
  >
    spinner
  </span>,
]
`;

exports[`Storyshots @mqs/react-server-components/components/LoadingIndicator Loading Indicator Size Example 1`] = `
[
  <span
    className="loading loading-spinner loading-xs"
  >
    xs
  </span>,
  <span
    className="loading loading-spinner loading-sm"
  >
    sm
  </span>,
  <span
    className="loading loading-spinner loading-md"
  >
    md
  </span>,
  <span
    className="loading loading-spinner loading-lg"
  >
    lg
  </span>,
]
`;
